src/update_agent/actor.rs

```diff
@@ -261,10 +257,14 @@ impl UpdateAgent {
 
 impl UpdateAgentInfo {
     /// Initialize the update agent.
-    async fn tick_initialize(&self, state: &mut UpdateAgentState) -> Result<(), ()> {
+    async fn tick_initialize(&self, state: &mut UpdateAgentState) {
         trace!("update agent in start state");
         if self.enabled {
-            self.register_as_driver().await?;
+            let register_outcome = self.register_as_driver().await;
```

Comment:
It looks like this method as well may be moved to `async` and stop returning a `Result`, perhaps?

Reply:
I've made it stop returning `Result`s, but I thought this is already an `async` function, or am I missing something...

Reply:
You are totally right, sorry for the confusion. I think I was locally reading by mistake a previous revision where `register_as_driver` was still using actix future types.